validator/find.go

```diff
@@ -0,0 +1,109 @@
+/*
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type root string
+
+// getDriverLibraryPath returns path to `libnvidia-ml.so.1` in the driver root.
+// The folder for this file is also expected to be the location of other driver files.
+func (r root) getDriverLibraryPath() (string, error) {
+	librarySearchPaths := []string{
+		"/usr/lib64",
+		"/usr/lib/x86_64-linux-gnu",
+		"/usr/lib/aarch64-linux-gnu",
+		"/lib64",
+		"/lib/x86_64-linux-gnu",
+		"/lib/aarch64-linux-gnu",
+	}
+
+	libraryPath, err := r.findFile("libnvidia-ml.so.1", librarySearchPaths...)
+	if err != nil {
+		return "", err
+	}
+
+	return libraryPath, nil
+}
+
+// getNvidiaSMIPath returns path to the `nvidia-smi` executable in the driver root.
+func (r root) getNvidiaSMIPath() (string, error) {
+	binarySearchPaths := []string{
+		"/usr/bin",
+		"/usr/sbin",
+		"/bin",
+		"/sbin",
```

Comment:
Did we maybe think about `/usr/local/bin` here? I am mostly curious about if there were maybe any counter-arguments.